lib/wit/parser.py

```diff
@@ -99,6 +99,33 @@ def err(msg):
 inspect_group.add_argument('--tree', action="store_true")
 inspect_group.add_argument('--dot', action="store_true")
 
+# ********** foreach subparser **********
+foreach_parser = subparsers.add_parser(
+    'foreach',
+    help='perform a command in each repository directory',
+    formatter_class=argparse.RawDescriptionHelpFormatter,
+    description="""
+Perform a command in each repository directory.
+The repository list is created by reading records contained in 'wit-lock.json'.
+
+If used, options --quiet and --continue-on-fail must be specified before the command.
+
+The following environment variables are exposed to the command:
+    WIT_REPO_NAME    repository name
+    WIT_REPO_PATH    path to repository
+    WIT_LOCK_SOURCE  inital source location of repository
+    WIT_LOCK_COMMIT  commit recorded in lockfile, actual repository contents could different
+    WIT_WORKSPACE    path to root of wit workspace""")
+
+foreach_parser.add_argument('--quiet', action='store_true',
+                            help='Supress printed lines indicating working repository')
+foreach_parser.add_argument('--continue-on-fail', action='store_true',
+                            help='run the command in each repository regardless of failures')
+
+# 'cmd' and 'args' eventually become one list, but this forces at least one input string
+foreach_parser.add_argument('cmd', help='command to run in each repository')
+foreach_parser.add_argument('args', nargs=argparse.REMAINDER, help='arguments for the command')
+
```

Comment:
Great job on the detailed description.

One minor clarification is that I was a little confused by what the environment variables mean when reading it. I would rephrase slightly to make it more clear that these are environment variables set by `wit` for each `command` rather than ones the user can set. I'm not sure on the best phrasing but here's a try:
```
Wit sets the following environment variables for each invocation of the command:
    WIT_REPO_NAME    repository name
    ...
```